compiler/noirc_evaluator/src/ssa/opt/normalize_value_ids.rs

```diff
@@ -57,12 +57,29 @@ struct IdMaps {
 
 impl Context {
     fn populate_functions(&mut self, functions: &BTreeMap<FunctionId, Function>) {
+        if functions.is_empty() {
+            return;
+        }
+
+        let old_purities = &functions.iter().next().unwrap().1.dfg.function_purities;
```

Comment:
```suggestion
        let Some(old_purities) = &functions.iter().next().map(|f| &f.1.dfg.function_purities) else {
            return;
        };
```
The lone `is_empty` check just felt a bit odd to me when I think every program should have an entry point at least.